### ISSUE TITLE
Script exists with no error after running iteration 0. Similar to issues #2483 and #767

### ISSUE THREAD

**Issue by jrsykes** [facebookresearch/detectron2#3747]
If you do not know the root cause of the problem, please post according to this template:

## Instructions To Reproduce the Issue:

Check https://stackoverflow.com/help/minimal-reproducible-example for how to ask good questions.
Simplify the steps to reproduce the issue using suggestions from the above link, and provide them below:

1. Full runnable code or full changes you made:
```
If making changes to the project itself, please use output of the following command:
git rev-parse HEAD; git diff

#####Link to Dateset:  https://drive.google.com/drive/folders/1Jb6xj4YRu55nzQhuWRijjJGvfe3yIQ3t?usp=sharing
#####

!pip install torch==1.9.0+cu102 torchvision==0.10.0+cu102 -f https://download.pytorch.org/whl/torch_stable.html

!pip install detectron2 -f https://dl.fbaipublicfiles.com/detectron2/wheels/cu102/torch1.9/index.html

!pip install -U git+https://github.com/albu/albumentations --no-cache-dir

exit(0)  

import torch, torchvision
print(torch.__version__, torch.cuda.is_available())
assert torch.__version__.startswith("1.9")   

import detectron2
from detectron2.utils.logger import setup_logger
setup_logger()


import numpy as np
import os, json, cv2, random
from google.colab.patches import cv2_imshow


from detectron2 import model_zoo
from detectron2.engine import DefaultPredictor
from detectron2.config import get_cfg
from detectron2.utils.visualizer import Visualizer
from detectron2.data import MetadataCatalog, DatasetCatalog

from google.colab import drive
drive.mount('/content/drive')

from pycocotools.coco import COCO
import numpy as np
import skimage.io as io
import matplotlib.pyplot as plt

dataDir='/content/drive/MyDrive/MaskR-CNN_annotated_subset/annotated_data_wholtree_subset/train/images'
dataType='val2017'
annFile='/content/drive/MyDrive/MaskR-CNN_annotated_subset/annotated_data_wholtree_subset/train/annotations/instances_default.json'.format(dataDir,dataType)
coco=COCO(annFile)

cats = coco.loadCats(coco.getCatIds())
nms=[cat['name'] for cat in cats]
print('COCO categories: \n{}\n'.format(' '.join(nms)))

catIds = coco.getCatIds(catNms=['Black pod rot', 'Witches broom disease', 'Frosty pod rot']);
imgIds = coco.getImgIds(catIds=catIds );
imgIds = coco.getImgIds(imgIds = [21])
img = coco.loadImgs(imgIds[np.random.randint(0,len(imgIds))])[0]

I = io.imread('%s/%s'%(dataDir,img['file_name']))
plt.axis('off')
plt.imshow(I)
plt.show()

plt.imshow(I); plt.axis('off')
annIds = coco.getAnnIds(imgIds=img['id'], catIds=catIds, iscrowd=None)
anns = coco.loadAnns(annIds)
coco.showAnns(anns)

from detectron2.data.datasets import register_coco_instances
register_coco_instances("my_dataset_train", {}, "/content/drive/MyDrive/MaskR-CNN_annotated_subset/annotated_data_wholtree_subset/train/annotations/instances_default.json", "/content/drive/MyDrive/MaskR-CNN_annotated_subset/annotated_data_wholtree_subset/train/images")
register_coco_instances("my_dataset_val", {}, "/content/drive/MyDrive/MaskR-CNN_annotated_subset/annotated_data_wholtree_subset/val/annotations/instances_default.json", "/content/drive/MyDrive/MaskR-CNN_annotated_subset/annotated_data_wholtree_subset/val/images")
register_coco_instances("my_dataset_test", {}, "/content/drive/MyDrive/MaskR-CNN_annotated_subset/annotated_data_wholtree_subset/test/annotations/instances_default.json", "/content/drive/MyDrive/MaskR-CNN_annotated_subset/annotated_data_wholtree_subset/test/images")

from detectron2.export import Caffe2Model, add_export_config, export_caffe2_model
from detectron2.modeling import build_model

cfg = get_cfg()
cfg.merge_from_file(model_zoo.get_config_file("COCO-InstanceSegmentation/mask_rcnn_R_50_FPN_3x.yaml"))
cfg.DATASETS.TRAIN = ("my_dataset_train",)
cfg.DATASETS.TEST = ("my_dataset_val",)
cfg.MODEL.WEIGHTS = model_zoo.get_checkpoint_url("COCO-InstanceSegmentation/mask_rcnn_R_50_FPN_3x.yaml")  
cfg.SOLVER.IMS_PER_BATCH = 2
cfg.MODEL.ROI_HEADS.BATCH_SIZE_PER_IMAGE = 42   
cfg.MODEL.SEM_SEG_HEAD.NUM_CLASSES = 3

model = build_model(cfg)

from detectron2.data import transforms as T
from detectron2.data import DatasetMapper   # the default mapper
from detectron2.data import build_detection_train_loader

dataloader = build_detection_train_loader(cfg,
   mapper=DatasetMapper(cfg, is_train=True, augmentations=[
      T.Resize((800, 800))
   ]), num_workers=2)

from detectron2.engine import SimpleTrainer
from detectron2.engine import HookBase
from detectron2.solver import get_default_optimizer_params


optimizer = torch.optim.Adam(get_default_optimizer_params(model),
               lr=1e-4)

os.makedirs(cfg.OUTPUT_DIR, exist_ok=True)

class CheckerHook(HookBase):
  def after_step(self):
    if self.trainer.iter % 100 == 0:
      print(f"Iteration {self.trainer.iter} complete")



trainer = SimpleTrainer(model, data_loader=dataloader, optimizer = optimizer) 
trainer.register_hooks([CheckerHook()])


trainer.train(start_iter=0, max_iter=100)

2. What exact command you run:

I run the final line in the script from within python: trainer.train(start_iter=0, max_iter=100)

3. __Full logs__ or other relevant observations:
```
[12/01 16:31:10 d2.engine.train_loop]: Starting training from iteration 0
/usr/local/lib/python3.7/dist-packages/torch/_tensor.py:575: UserWarning: floor_divide is deprecated, and will be removed in a future version of pytorch. It currently rounds toward 0 (like the 'trunc' function NOT 'floor'). This results in incorrect rounding for negative values.
To keep the current behavior, use torch.div(a, b, rounding_mode='trunc'), or for actual floor division, use torch.div(a, b, rounding_mode='floor'). (Triggered internally at  /pytorch/aten/src/ATen/native/BinaryOps.cpp:467.)
  return torch.floor_divide(self, other)
/usr/local/lib/python3.7/dist-packages/torch/nn/functional.py:718: UserWarning: Named tensors and all their associated APIs are an experimental feature and subject to change. Please do not use them for anything important until they are released as stable. (Triggered internally at  /pytorch/c10/core/TensorImpl.h:1156.)
  return torch.max_pool2d(input, kernel_size, stride, padding, dilation, ceil_mode)
Iteration 0 complete

After running %debug I get:
ERROR:root:No traceback has been produced, nothing to debug.

```

## Expected behavior:

If there are no obvious crash in "full logs" provided above,
please tell us the expected behavior.

If you expect a model to converge / work better, we do not help with such issues, unless
a model fails to reproduce the results in detectron2 model zoo, or proves existence of bugs.

I expect the trainer to run but instead it runs for a couple of minutes and then exits after the the first iteration with no error code (except for the above warnings).

This problem is similar to those in issues #2483 and #767. However, decreasing the dataloader workers to 1 or 0 did not solve the problem.

## Environment:

Paste the output of the following command:
```
wget -nc -nv https://github.com/facebookresearch/detectron2/raw/master/detectron2/utils/collect_env.py && python collect_env.py
```
This command gave: ERROR 404: Not Found.

google colab
sys.platform: linux
torch version 1.9
CUDA Version: 11.2
GPU RAM: 11441MiB
GPU: NVIDIA Tesla K80
CUDA available True
Python version: 3.7.12
Pillow 7.1.2
numpy: 1.19.5
cv2: 4.5.4

%env output:

Install detectron2
[ ]
#!pip install pyyaml==5.1
# This is the current pytorch version on Colab. Uncomment this if Colab changes its pytorch version
!pip install torch==1.9.0+cu102 torchvision==0.10.0+cu102 -f https://download.pytorch.org/whl/torch_stable.html

# Install detectron2 that matches the above pytorch version
# See https://detectron2.readthedocs.io/tutorials/install.html for instructions
!pip install detectron2 -f https://dl.fbaipublicfiles.com/detectron2/wheels/cu102/torch1.9/index.html

!pip install -U gi

[22]
0s
# check pytorch installation: 
import torch, torchvision
#print(torch.__version__, torch.cuda.is_available())
#assert torch.__version__.startswith("1.9")   # please manually install torch 1.9 if Colab changes its default version
#!nvidia-smi

%env
{'CLICOLOR': '1',
 'CLOUDSDK_CONFIG': '/content/.config',
 'CLOUDSDK_PYTHON': 'python3',
 'COLAB_GPU': '1',
 'CUDA_VERSION': '11.1.1',
 'CUDNN_VERSION': '8.0.5.39',
 'DATALAB_SETTINGS_OVERRIDES': '{"kernelManagerProxyPort":6000,"kernelManagerProxyHost":"172.28.0.3","jupyterArgs":["--ip=\\"172.28.0.2\\""],"debugAdapterMultiplexerPath":"/usr/local/bin/dap_multiplexer","enableLsp":true}',
 'DEBIAN_FRONTEND': 'noninteractive',
 'ENV': '/root/.bashrc',
 'GCE_METADATA_TIMEOUT': '0',
 'GCS_READ_CACHE_BLOCK_SIZE_MB': '16',
 'GIT_PAGER': 'cat',
 'GLIBCPP_FORCE_NEW': '1',
 'GLIBCXX_FORCE_NEW': '1',
 'HOME': '/root',
 'HOSTNAME': '5a9b9f4b686b',
 'JPY_PARENT_PID': '62',
 'LANG': 'en_US.UTF-8',
 'LAST_FORCED_REBUILD': '20211116',
 'LD_LIBRARY_PATH': '/usr/local/lib/python3.7/dist-packages/cv2/../../lib64:/usr/lib64-nvidia',
 'LD_PRELOAD': '/usr/lib/x86_64-linux-gnu/libtcmalloc.so.4',
 'LIBRARY_PATH': '/usr/local/cuda/lib64/stubs',
 'MPLBACKEND': 'module://ipykernel.pylab.backend_inline',
 'NCCL_VERSION': '2.7.8',
 'NO_GCE_CHECK': 'True',
 'NVIDIA_DRIVER_CAPABILITIES': 'compute,utility',
 'NVIDIA_REQUIRE_CUDA': 'cuda>=11.1 brand=tesla,driver>=418,driver<419 brand=tesla,driver>=440,driver<441 brand=tesla,driver>=450,driver<451',
 'NVIDIA_VISIBLE_DEVICES': 'all',
 'OLDPWD': '/',
 'OPENCV_OPENCL_RUNTIME': 'disabled',
 'PAGER': 'cat',
 'PATH': '/usr/local/nvidia/bin:/usr/local/cuda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/tools/node/bin:/tools/google-cloud-sdk/bin:/opt/bin',
 'PWD': '/',
 'PYDEVD_USE_FRAME_EVAL': 'NO',
 'PYTHONPATH': '/env/python',
 'PYTHONWARNINGS': 'ignore:::pip._internal.cli.base_command',
 'SHELL': '/bin/bash',
 'SHLVL': '0',
 'TBE_CREDS_ADDR': '172.28.0.1:8008',
 'TBE_EPHEM_CREDS_ADDR': '172.28.0.1:8009',
 'TERM': 'xterm-color',
 'TF_FORCE_GPU_ALLOW_GROWTH': 'true',
 'USE_EPHEM': '0',
 '__EGL_VENDOR_LIBRARY_DIRS': '/usr/lib64-nvidia:/usr/share/glvnd/egl_vendor.d/'}

If your issue looks like an installation issue / environment issue,
please first check common issues in https://detectron2.readthedocs.io/tutorials/install.html#common-installation-issues

**Comment by jrsykes** [comment]
Solved! I had it set to 100 iterations while I debugged the code but apparently the minimum is 101 iterations... of course. Works fine now.
